hs_core/views/resource_rest_api.py

```diff
@@ -517,7 +518,9 @@ def create(self, request, *args, **kwargs):
                     keywords=keywords,
                     metadata=metadata,
                     extra_metadata=extra_metadata,
-                    files=files
+                    files=files,
+                    source_names=reference_files,
+                    is_file_reference=True if len(reference_files) > 0 else False
```

Comment:
@ismagilani checking ```len(reference_files)``` would require the default value for reference_files on line 475 above to be ```[]``` rather than ```None```. Otherwise, this will raise an exception.

Reply:
Since it's StringListField in the model, I think we are safe to assume it will be [] not None.  Do you agree?

Reply:
I agree. It'd be great if you can test it out to be sure, though. But I know for sure ```len()``` does not apply to a None object.

Reply:
Surely, let me run a couple of tests.